polkadot/node/subsystem-bench/src/availability/mod.rs

```diff
@@ -82,72 +106,259 @@ fn build_overseer(
+						.expect("candidate was generated previously; qed");
+					gum::warn!(target: LOG_TARGET, ?candidate_hash, candidate_index, "Candidate mapped to index");
+
+					let chunk: ChunkResponse =
+						self.chunks.get(*candidate_index).unwrap()[validator_index].clone().into();
+					let response = Ok((
+						ChunkFetchingResponse::from(Some(chunk)).encode(),
+						ProtocolName::Static("dummy"),
+					));
+
+					if let Err(err) = outgoing_request.pending_response.send(response) {
+						gum::error!(target: LOG_TARGET, ?err, "Failed to send `ChunkFetchingResponse`");
+					}
+
+					None
+				},
+				Requests::AvailableDataFetchingV1(outgoing_request) => {
+					let candidate_hash = outgoing_request.payload.candidate_hash;
+					let candidate_index = self
+						.candidate_hashes
+						.get(&candidate_hash)
+						.expect("candidate was generated previously; qed");
+					gum::debug!(target: LOG_TARGET, ?candidate_hash, candidate_index, "Candidate mapped to index");
+
+					let available_data = self.available_data.get(*candidate_index).unwrap().clone();
+
+					let response = Ok((
+						AvailableDataFetchingResponse::from(Some(available_data)).encode(),
+						ProtocolName::Static("dummy"),
+					));
+					outgoing_request
+						.pending_response
+						.send(response)
+						.expect("Response is always sent succesfully");
+					None
+				},
+				_ => Some(NetworkMessage::RequestFromNode(peer, request)),
+			},
+
+			message => Some(message),
+		}
+	}
+}
+
+/// Takes a test configuration and uses it to create the `TestEnvironment`.
 pub fn prepare_test(
 	config: TestConfiguration,
 	state: &mut TestState,
-) -> (TestEnvironment, ProtocolConfig) {
+) -> (TestEnvironment, Vec<ProtocolConfig>) {
 	prepare_test_inner(config, state, TestEnvironmentDependencies::default())
 }
 
 fn prepare_test_inner(
 	config: TestConfiguration,
 	state: &mut TestState,
 	dependencies: TestEnvironmentDependencies,
-) -> (TestEnvironment, ProtocolConfig) {
+) -> (TestEnvironment, Vec<ProtocolConfig>) {
 	// Generate test authorities.
 	let test_authorities = config.generate_authorities();
 
-	let runtime_api = runtime_api::MockRuntimeApi::new(config.clone(), test_authorities.clone());
+	let mut candidate_hashes: HashMap<H256, Vec<CandidateReceipt>> = HashMap::new();
+
+	// Prepare per block candidates.
+	// Genesis block is always finalized, so we start at 1.
+	for block_num in 0..=config.num_blocks {
+		for _ in 0..config.n_cores {
+			candidate_hashes
+				.entry(Hash::repeat_byte(block_num as u8))
+				.or_default()
+				.push(state.next_candidate().expect("Cycle iterator"))
+		}
 
-	let av_store =
-		av_store::MockAvailabilityStore::new(state.chunks.clone(), state.candidate_hashes.clone());
+		// First candidate is our backed candidate.
+		state.backed_candidates.push(
+			candidate_hashes
+				.get(&Hash::repeat_byte(block_num as u8))
+				.expect("just inserted above")
+				.get(0)
+				.expect("just inserted above")
+				.clone(),
+		);
```

Comment:
I am not sure I understand correctly, but `backed_candidates` is a `vec` ( per relay block candidates backed by the node under test) and it stores CandidateReceipts..

Reply:
right, in my mind vec is also a map from an integer to smth :D 

i am suggesting based on my reading of the code, that we could only store the CandidateHash in the `backed_candidates` instead of the full receipt and then  use the candidate index to fetch the receipt from the `candidate_receipt_templates`, if that makes sense.